### ISSUE TITLE
NUnit2021 Should not raise for UsingPropertiesComparer

### ISSUE THREAD

**Issue by manfred-brands** [nunit/nunit.analyzers#857]
The latest changes to NUnit's properties comparer allows comparing the common properties of different types.

```csharp
        [Test]
        public void AssertThatDifferentTypesCanBeCompared()
        {
            var instanceC = new ClassWithSomeToleranceAwareMembers(1, 1.1, "1.1", null);
            var instanceS = new StructWithSomeToleranceAwareMembers(1, 1.1, "1.1", SomeEnum.Two);

            Assert.That(instanceS, Is.EqualTo(instanceC).UsingPropertiesComparer(
                c => c.CompareOnlyCommonProperties()));
        }
```

`UsingPropertiesComparer` also supports type mapping:

```csharp
            Assert.That(usPerson, Is.EqualTo(person).UsingPropertiesComparer(
                c => c.Map<USPerson>(x => x.Address, y => y.USAddress)
                      .Map<Address, USAddress>(x => x.AreaCode, y => y.ZipCode)
                      .Map<Address>(x => x.Country, "U.S.A.")));
```
